payments_service/solitude/__init__.py

```diff
@@ -13,12 +13,37 @@
 
 
 def api():
-    conn = API(settings.SOLITUDE_URL)
+    conn = SolitudeAPI(settings.SOLITUDE_URL)
     conn.activate_oauth(settings.SOLITUDE_KEY,
                         settings.SOLITUDE_SECRET)
     return conn
 
 
+def url_parser(url):
+    """
+    Curling URL parser for resources with numeric primary keys.
+
+    It helps convert URLs into Slumber resources like:
+
+    /service/category/thing/1/ -> service.category.thing(1)
+    """
+    # Split all URL parts ignoring empties (from slashes).
+    parts = [p for p in url.split('/') if p != '']
+
+    pk = None
+    if parts[-1].isdigit():
+        pk = parts.pop()
+
+    return parts, pk
```

Comment:
oh yeah, I'll merge the two. Looks like that one doesn't handle the no-primary-key case.